lib/cache/resource.js

```diff
@@ -0,0 +1,122 @@
+'use strict';
+
+var _ = require('lodash');
+var MemoryCache = require('./memory');
+
+function ResourceCache(cacheId, options) {
+    this._cache = new MemoryCache(cacheId, options);
+}
+
+ResourceCache.prototype.get = function(cacheKey) {
+    return getFromCache.call(this, cacheKey);
+};
+
+function getFromCache(cacheKey) {
+    var entry = _.cloneDeep(this._cache.get(cacheKey));
+
+    // If parent or nested resources not in cache then return undefined.
```

Comment:
No, the opposite. If the primaryPhoto resource had an expiration shorter than the main user data resource, it'd expire the main user data resource, forcing us to go re-get the main user data resource, because the primary photo resource is expired.